### PR TITLE
Replace Travis with Github Actions

### PR DESCRIPTION
**Pull request by ndanielsen**
This PR is to replace Travis CI with Github Actions.

CI is working nicely on Ubuntu, windows, and macos running python 3.7-3.8 and conda only working with ubuntu at the moment. Running the test suite with conda for macos and windows was timing out, but can likely be fixed with some investigation.

Slack hook integration working with a webhook to a non-yb slack group. [Result Link](https://github.com/ndanielsen/yellowbrick/actions/runs/587188664)

In order to get conda to work, I had to downgrade `pytest-spec`.

Still to do:

- [x] Get Conda working (maybe find a work around on pytest-spec)
- [x] Get Coveralls working (or migrate to codecov like many other OSS projects)
- [x] Add slack notification webhook
- [ ] Remove travis stuff

**Post Merge TODOs**

- Add the `SLACK_WEBHOOK` to secrets. 
- Add codecov for this repo

See pytest-spec issue:
https://github.com/pchomik/pytest-spec/issues/41

### CODE REVIEW

File: tests/requirements.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ cycler==0.10.0
 pytest==6.1.1
 pytest-cov==2.10.1
 pytest-flakes==4.0.2
-pytest-spec==3.0.4
+pytest-spec>=2.0.0
```
Comment (ndanielsen) [inline tests/requirements.txt:17]: downgrade to allow conda to work

File: setup.cfg [hunk 2]
```diff
@@ -10,7 +10,7 @@ tests = True
 test=pytest
 
 [tool:pytest]
-addopts = --verbose --cov=yellowbrick --flakes --spec
+addopts = --verbose --cov=yellowbrick --flakes --spec --cov-report=xml --cov-report term
```
Comment (ndanielsen) [inline setup.cfg:13]: allow codecov integration to work

**Comment by pdamodaran** [comment]
@ndanielsen - Thanks for submitting this pull request and providing us a GitHub Actions solution as an alternative to Travis. We have made the determination to migrate from Travis to GitHub Actions and are happy to use your solution. I made some minor changes to this PR.

File: tests/requirements.txt (hunk 1 above)
Comment (pdamodaran) [reply tests/requirements.txt:17]: thanks for making this fix!

**Comment by ndanielsen** [comment]
Thanks @pdamodaran. Happy to keep working on this as there are a few other changes to make and was waiting on any feedback.

**Comment by ndanielsen** [comment]
happy to continue contributing and other maintenance tasks if there is a need.

**Comment by pdamodaran** [comment]
> Thanks @pdamodaran. Happy to keep working on this as there are a few other changes to make and was waiting on any feedback.

@ndanielsen - feel free to suggest any changes that need to be made.  The ones I made were cosmetic.

**Comment by ndanielsen** [comment]
Thanks for the merge. I wasn't ready to pick up the other smaller issues at the moment like the statuses badges and doing a documentation scale to remove references if they are present. Are you going to do it or should I pick that up?

**Comment by ndanielsen** [comment]
Also TODOs:

- enabling github actions on this repo
- setting up the slack token access
- setting up codecov

**Comment by ndanielsen** [comment]
Again happy to help if desired

**Pull request merged**